### PR TITLE
+doc,htc #18610 document using singleRequest in Actors

### LINKED ISSUE #18610: Akka HTTP, docs: Document using singleRequest() with Actors
We have `ImplicitMaterializer` and `pipeTo`, however it's not documented on the HTTP page.
Should be shown as simple example.

Based on feedback from: http://stackoverflow.com/questions/32840748/rest-client-in-akka-with-java/32854118#32854118

### PR DESCRIPTION
**Pull request by ktoso**
Resolves #18610

### CODE REVIEW

**Comment by ktoso** [comment]
Small docs update to address questions coming up on stack overflow.

**Comment by ktoso** [comment]
PLS BUILD

**Comment by ktoso** [comment]
Failure was ssh not having permissions to issue commands to the slave node hm...

**Comment by ktoso** [comment]
Please review :)

**Comment by ktoso** [comment]
Sanity check this PR before I merge please :)
It bumps compiler level to JDK8 for samples.

// cc @patriknw @2m @drewhk

File: akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java [hunk 1]
```diff
@@ -69,4 +73,25 @@ public void testSingleRequestExample() {
...
+      //#single-request-in-actor-example
+    class Myself extends AbstractActor {
+      final Http http = Http.get(context().system());
+      final ExecutionContextExecutor dispatcher = context().dispatcher();
+      final ActorMaterializer materializer = ActorMaterializer.create(context().system());
+
+      public Myself() {
+        receive(ReceiveBuilder
+         .match(String.class, url -> {
+           Patterns.<HttpResponse>pipe(fetch(url), dispatcher).to(self());
```
Comment (patriknw) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java]: is the type annotation `<HttpResponse>` need here ?

File: akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java [hunk 2]
```diff
@@ -69,4 +73,25 @@ public void testSingleRequestExample() {
...
   }
-}
\ No newline at end of file
+
+  static
+      //#single-request-in-actor-example
+    class Myself extends AbstractActor {
+      final Http http = Http.get(context().system());
+      final ExecutionContextExecutor dispatcher = context().dispatcher();
+      final ActorMaterializer materializer = ActorMaterializer.create(context().system());
```
Comment (patriknw) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java]: don't we have something for creating materializer within another actor?

File: akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java (hunk 1 above)
Comment (ktoso) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java]: AFAIR yes, I wasn't able to make it work without :(

File: akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java (hunk 2 above)
Comment (patriknw) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java]: like `ImplicitMaterializer` but for java?

Comment (ktoso) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java]: only in Scala, where you can `with ImplicitMaterializer`

Comment (ktoso) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java]: I'll give it a look or maybe add something for java then hm hm

File: akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java (hunk 1 above)
Comment (patriknw) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java]: strange, it's not needed here: http://doc.akka.io/docs/akka/2.4.0/java/lambda-actors.html#Ask__Send-And-Receive-Future

**Comment by patriknw** [comment]
LGTM

File: akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java (hunk 2 above)
Comment (ktoso) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java]: Syntax wise we can already `ActorMaterializer.create(context())`, I'll use that

File: akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java (hunk 1 above)
Comment (ktoso) [inline akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java]: Hah, intellij can't figure it out though `javac` can.
It infers properly, removed the explicit type.

// feels like early years of scala when you couldn't trust your IDE, now the same happens to java... :wink:

**Comment by ktoso** [comment]
Will merge after validation.

**Pull request merged**